benchmark/utils.go

```diff
@@ -20,16 +20,87 @@ import (
 	"encoding/csv"
 	"encoding/json"
 	"errors"
+	"fmt"
 	"io"
 	"os"
 	"os/exec"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/contrib/nvidia"
+	"github.com/containerd/containerd/oci"
+	runtimespec "github.com/opencontainers/runtime-spec/specs-go"
 )
 
+const testContainerID = "TEST_RUN_CONTAINER"
+
 type ImageDescriptor struct {
-	ShortName            string `json:"short_name"`
-	ImageRef             string `json:"image_ref"`
-	SociIndexManifestRef string `json:"soci_index_manifest_ref"`
-	ReadyLine            string `json:"ready_line"`
+	ShortName            string       `json:"short_name"`
+	ImageRef             string       `json:"image_ref"`
+	SociIndexManifestRef string       `json:"soci_index_manifest_ref"`
+	ReadyLine            string       `json:"ready_line"`
+	TimeoutSec           int64        `json:"timeout_sec"`
+	ImageOptions         ImageOptions `json:"options"`
+}
+
+func (i *ImageDescriptor) Timeout() time.Duration {
+	if i.TimeoutSec <= 0 {
+		return 180 * time.Second
+	}
+	return time.Duration(i.TimeoutSec) * time.Second
+}
+
+// ImageOptions contains image-specific options needed to run the tests
+type ImageOptions struct {
```

Comment:
Given image customization is tied to JSON, should we consider removing CSV entirely? We can track that in a separate issue.

Reply:
Yeah, that was generally the direction I wanted to go. I'll create an issue.

Reply:
https://github.com/awslabs/soci-snapshotter/issues/946